ytdlbot/sp_downloader.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/local/bin/python3
+# coding: utf-8
+
+# ytdlbot - sp_downloader.py
+# 3/16/24 16:32
+#
+
+__author__ = "Benny <benny.think@gmail.com>, SanujaNS <sanujas@sanuja.biz>"
+
+import pathlib
+import logging
+import traceback
+import re
+import requests
+from tqdm import tqdm
+import json
+from bs4 import BeautifulSoup
+from urllib.parse import parse_qs, urlparse
+import filetype
+import yt_dlp as ytdl
+
+from config import (
+    ENABLE_ARIA2,
+    IPv6,
+)
+
+user_agent = (
+    "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.6261.128 Safari/537.36"
+)
+
+
+def sp_dl(url: str, tempdir: str):
+    """Specific link downloader"""
+    domain = urlparse(url).hostname
+    if not any(
+        x in domain
+        for x in [
+            "www.instagram.com",
+            "pixeldrain.com",
+            "mediafire.com",
+        ]
+    ):
+        return False
+    if "www.instagram.com" in domain:
+        return instagram(url, tempdir)
+    elif "pixeldrain.com" in domain:
+        return pixeldrain(url, tempdir)
+    elif "www.xasiat.com" in domain:
+        return xasiat(url, tempdir)
+
+
+def sp_ytdl_download(url: str, tempdir: str):
```

Comment:
the code in this function appears to be much duplicated comparing with the one in `downloader.py`. Anyway we can avoid those duplicates?

Reply:
Yes, It's the same function striped some parts. It would be great if we can call the function in `downloader.py` but I can't think of a way without it looping itself because some URLs(even tho it's direct URL) include parts used in handling function(sp_dl).